### PR TITLE
DOC: Comparison with SAS

### LINKED ISSUE #11075: DOC: Comparison with SAS
I don't really use SAS any more, but I've got some SAS -> pandas notes I was thinking of expanding to a doc section (like the R & SQL ones) as long as people don't think it's too out of scope for the pandas docs.

### PR DESCRIPTION
**Pull request by chris-b1**
closes #11075
xref #4052 

There doesn't seem to be a straightforward way to syntax highlight the SAS code, so it will format as generic code blocks.

### CODE REVIEW

**Comment by jreback** [comment]
lots of lexers here: http://pygments.org/docs/lexers/#lexers-for-the-r-s-languages

isn't it actual S syntax?

**Comment by chris-b1** [comment]
No, SAS has its own syntax.

**Comment by jreback** [comment]
can you render and post this page

**Comment by jreback** [comment]
can also add to the highlites section in v0.17.0 (and add to the release.rst the same line).

**Comment by chris-b1** [comment]
Here's the rendered output - for some reason in-line code segments don't get rendered correctly locally for me (on all doc pages).

docs

**Comment by jreback** [comment]
@chris-b1 looks pretty good.

I need a more full look, but @jorisvandenbossche ?

File: doc/source/comparison_with_sas.rst [hunk 1]
```diff
@@ -0,0 +1,604 @@
...
+   tips.head()
+
+
+Like ``PROC IMPORT``, ``read_csv`` can take a number of parameters to specify
+how the data should be parsed.  For example, if the data was instead tab delimited,
+and did not have column names, the pandas command would be:
+
+.. code-block:: python
+
+   tips = pd.read_csv('tips.csv', delim='\t', header=False)
```
Comment (jorisvandenbossche) [inline doc/source/comparison_with_sas.rst]: delim -> sep (delimiter is a valid keyword alternative to sep, but sep is the preferred one (listed first in the docstring))

File: doc/source/comparison_with_sas.rst [hunk 2]
```diff
@@ -0,0 +1,604 @@
...
+
+SAS provides ``PROC IMPORT`` to read csv data into a data set.
+
+.. code-block:: none
+
+   proc import datafile='tips.csv' dbms=csv out=tips replace;
+       getnames=yes;
+   run;
+
+The pandas method is ``read_csv``, which works similarly.
```
Comment (jorisvandenbossche) [inline doc/source/comparison_with_sas.rst]: Can you use ``` :func:`read_csv` ``` here? then it becomes a link to the docstring page (which is nice)

File: doc/source/comparison_with_sas.rst [hunk 3]
```diff
@@ -0,0 +1,604 @@
...
+part of pandas to understand, for this comparison we will essentially ignore the
+``Index`` and just treat the ``DataFrame`` as a collection of columns. Please
+see the :ref:`indexing documentation<indexing>` for much more on how to use an
+``Index`` effectively.
+
+
+Data Input / Output
+-------------------
+
+Reading keyed-in values:
```
Comment (jorisvandenbossche) [inline doc/source/comparison_with_sas.rst]: 'keyed-in' values sounds a bit strange to me. What do you mean with this?

Comment (jorisvandenbossche) [inline doc/source/comparison_with_sas.rst]: very minor comment, but using a `:` in a title looks a bit strange to me personally in the rendered output

Comment (chris-b1) [inline doc/source/comparison_with_sas.rst]: I was struggling for the right term - "keyed-in" meaning the data is specified in program, not from an external source.

File: doc/source/comparison_with_sas.rst [hunk 4]
```diff
@@ -0,0 +1,604 @@
...
+
+Sorting in SAS is accomplished via ``PROC SORT``
+
+.. code-block:: none
+
+   proc sort data=tips;
+       by sex total_bill;
+   run;
+
+pandas objects have a ``sort_values`` method, which
```
Comment (jorisvandenbossche) [inline doc/source/comparison_with_sas.rst]: same here as for read_csv (in this case ``` :meth:`~DataFrame.sort_values` ``` is best, the `~` makes you only see the sort_values and not the 'DataFrame')

File: doc/source/comparison_with_sas.rst [hunk 5]
```diff
@@ -0,0 +1,604 @@
...
+       merge df1(in=a) df2(in=b);
+
+       if a and b then output inner_join;
+       if a then output left_join;
+       if b then output right_join;
+       if a or b then output outer_join;
+   run;
+
+pandas DataFrames have a merge method, which provides
+similar funtionality.  Note that the data does not have
```
Comment (jorisvandenbossche) [inline doc/source/comparison_with_sas.rst]: funtionality -> functionality

**Comment by jorisvandenbossche** [comment]
@chris-b1 Really cool!!

just added some minor comments

File: doc/source/comparison_with_sas.rst (hunk 3 above)
Comment (jorisvandenbossche) [inline doc/source/comparison_with_sas.rst]: Maybe something like 'Constructing a DataFrame' ? (as it is not really 'reading' data)

Comment (chris-b1) [inline doc/source/comparison_with_sas.rst]: Thanks, I think I'll change it to 'Constructing a DataFrame from Values` - SAS has very limited structures - the code is basically like `StringIO`, which is I was using the term 'reading'.

Comment (jorisvandenbossche) [inline doc/source/comparison_with_sas.rst]: yep, that sounds good

**Comment by chris-b1** [comment]
@jorisvandenbossche - pushed changes for your comments, thanks for taking a look.  Rendered output above is also updated.

File: doc/source/comparison_with_sas.rst [hunk 6]
```diff
@@ -0,0 +1,604 @@
...
+   tips = pd.read_csv('tips.csv', sep='\t', header=False)
+
+   # alternatively, read_table is an alias to read_csv with tab delimiter
+   tips = pd.read_table('tips.csv', header=False)
+
+In addition to text/csv, pandas supports a variety of other data formats
+such as Excel, HDF5, and SQL databases.  These are all read via a ``pd.read_*``
+function.  See the :ref:`IO documentation<io>` for more details.
+
+Exporting data:
```
Comment (jreback) [inline doc/source/comparison_with_sas.rst]: don't use ``:`` for sub-headings / headings...

File: doc/source/comparison_with_sas.rst [hunk 7]
```diff
@@ -0,0 +1,604 @@
...
+       set outer_join;
+       if value_x = .;
+   run;
+
+Which doesn't work in in pandas.  Instead, the ``pd.isnull`` or ``pd.notnull`` functions
+should be used for comparisons.
+
+.. ipython:: python
+
+   # doesn't work, NaN != NaN
```
Comment (jreback) [inline doc/source/comparison_with_sas.rst]: I wouldn't even show this, just show using ``.isnull`` and co

File: doc/source/comparison_with_sas.rst [hunk 8]
```diff
@@ -0,0 +1,604 @@
...
+
+
+pandas ``groubpy`` provides a ``transform`` mechanism that allows
+these type of operations to be succinctly expressed in one
+operation.
+
+.. ipython:: python
+
+   gb = tips.groupby('smoker')['total_bill']
+   tips['adj_total_bill'] = gb.transform(lambda x: x - x.mean())
```
Comment (jreback) [inline doc/source/comparison_with_sas.rst]: canoncially

``tips['total_bill'] - gb.transformat('mean')``

**Comment by jreback** [comment]
@chris-b1 just a couple of minor edits / suggestions.

really nice!

ping when ready for merge.

**Comment by chris-b1** [comment]
Thanks!  @jreback pushed changes for your notes, should be good to merge.

**Comment by jreback** [comment]
awesome. pls have a look once they are built and do a follow-up if needed!

**Pull request merged**